internal/smartlink/smartlink.go

```diff
@@ -24,12 +24,23 @@ func LinkContents(src, dest string) error {
 		return errs.Wrap(err, "Could not resolve src and dest paths")
 	}
 
+	if visited == nil {
+		visited = make(map[string]bool)
+	}
+	if _, exists := visited[src]; exists {
+		// We've encountered a recursive link. This is most often the case when the resolved src has
+		// already been visited. In that case, just link the dest to the src (which may be a directory;
+		// this is fine).
+		return linkFile(src, dest)
+	}
+	visited[src] = true
```

Comment:
I don't think we need this check here. I think just the one on the Link function already covers our use-case.